File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        int n = size(costs);
        int ct = 0;
        sort(begin(costs), end(costs));

        for (int i = 0; i < n; i++) {
            if (coins - costs[i] < 0)
                break;
            coins -= costs[i];
            ct++;

        }
        return ct;
    }
};

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.ts
function maxIceCream(costs: number[], coins: number): number {
    let n: number = costs.length;
    let ct: number = 0;
    costs.sort((a, b) => {
        return a - b;
    });
    for (let i: number = 0; i < n; i++) {
        let diff: number = coins - costs[i];
        if (diff < 0)
            break;

        coins -= costs[i];
        ct++;
    }
    return ct;
};